platform-tests/src/acceptance/http_edge_cases_test.go

```diff
@@ -0,0 +1,205 @@
+)
+
+const (
+	letters     = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	utfchars    = "スタック・オーバーフロー はプログラマ"
+	paramLength = 8
+	queryLength = 4096
+	requesttext = "slow response"
+)
+
+var (
+	appName        string
+	numberOfParams = (queryLength / paramLength)
+	queryString    string
+	parameters     url.Values
+)
+
+var _ = Describe("HTTP edge cases", func() {
+	Describe("Using dora", func() {
+
+		BeforeEach(func() {
+			appName = generator.PrefixedRandomName("CATS-APP-DORA-")
+			Expect(cf.Cf(
+				"push", appName,
+				"-b", config.RubyBuildpackName,
+				"-p", "../../../../cf-release/src/github.com/cloudfoundry/cf-acceptance-tests/assets/dora",
+				"-d", config.AppsDomain,
+				"-i", "1",
+				"-m", "256M",
+			).Wait(CF_PUSH_TIMEOUT)).To(Exit(0))
+		})
+
+		It("should serve request and response bodies of increasing sizes", func() {
+			for responsekB := 1; responsekB <= 200; responsekB += 10 {
+				By(fmt.Sprintf("response size of %d kB", responsekB))
+				response := helpers.CurlAppWithTimeout(
+					appName,
+					fmt.Sprintf("/largetext/%d", responsekB),
+					5*time.Second,
+				)
+				Expect(response).To(HaveLen(responsekB * int(KILOBYTE)))
+			}
+		})
+	})
+
+	Describe("Using http_tester", func() {
+		var appName string
+		var httpClient http.Client
+
+		BeforeEach(func() {
+			appName = generator.PrefixedRandomName("CATS-APP-HTTP-TESTER-")
+			Expect(cf.Cf(
+				"push", appName,
+				"-b", config.GoBuildpackName,
+				"-p", "../../example-apps/http_tester",
+				"-d", config.AppsDomain,
+				"-c", "./bin/debug_app; sleep 1; echo 'done'",
+			).Wait(CF_PUSH_TIMEOUT)).To(Exit(0))
+			rand.Seed(time.Now().UnixNano())
+			tr := &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			}
+			httpClient = http.Client{Transport: tr}
+		})
+
+		It("should accept large request body sizes", func() {
+			for requestkB := 1; requestkB <= 200; requestkB += 10 {
+				By(fmt.Sprintf("request size of %d kB", requestkB))
+				buffer := new(bytes.Buffer)
+				file, err := os.Open("/dev/zero")
+				Expect(err).NotTo(HaveOccurred())
+				defer file.Close()
+
+				copied, err := io.CopyN(buffer, file, int64(requestkB*int(KILOBYTE)))
+				Expect(err).NotTo(HaveOccurred())
+				fmt.Fprintf(GinkgoWriter, "Successfully copied %d bytes\n", copied)
+
+				appUri := helpers.AppUri(appName, "/body-size")
+
+				req, err := http.NewRequest("POST", appUri, buffer)
+				Expect(err).NotTo(HaveOccurred())
+
+				response, err := httpClient.Do(req)
+				Expect(err).NotTo(HaveOccurred())
+				returnedBody, err := ioutil.ReadAll(response.Body)
+				Expect(err).NotTo(HaveOccurred())
+				returnedBodySize, err := strconv.Atoi(string(returnedBody))
+				Expect(err).NotTo(HaveOccurred())
+				Expect(returnedBodySize).Should(Equal(requestkB * int(KILOBYTE)))
+			}
+		})
+
+		It("should accept large request header sizes", func() {
+			for headerkB := 1; headerkB <= 7; headerkB += 1 {
```

Comment:
It's not a limit in the HTTP spec rather a limit imposed by the webserver 8KB is the generally accepted max with IIS upping it to 16KB. This was verified during developing the test.

https://stackoverflow.com/questions/686217/maximum-on-http-header-values